modules/core/03-connection/keeper/verify.go

```diff
@@ -364,6 +364,44 @@ func (k Keeper) VerifyNextSequenceRecv(
 	return nil
 }
 
+// VerifyChannelUpgradeSequence verifies a proof of the upgrade sequence number to be
+// used during channel upgrades.
+func (k Keeper) VerifyChannelUpgradeSequence(
+	ctx sdk.Context,
+	connection exported.ConnectionI,
+	height exported.Height,
+	proof []byte,
+	portID,
```

Comment:
should these be `counterpartyPortID` and `counterpartyChannelID` as per the spec? just to be explicit